### PR TITLE
HOTT-909: Cookies banner reappears

### PR DESCRIPTION
**Pull request by jebw**
### Jira link

[HOTT-909](https://transformuk.atlassian.net/browse/HOTT-909)

### What?

I have added/removed/altered:

- [x] Added a `Cookies::Policy` model to control access to the policy cookie - this provides replacements for the various methods in `CookiesHelper` for reading and writing to the cookie
- [x] Added a `Cookies::PoliciesController` to manipulate that cookie, this takes over from `PagesController#tariff_cookies`, `PagesController#update_cookies`, and `CookiesConsentController`
- [x] Added a `Cookies::HideConfirmationsController` to manipulate the 'hide confirmation' cookie - this takes over from `CookiesConsentController#add_seen_confirmation`
- [x] Added a `Cookies` feature spec to verify the different paths a user can follow

### Why?

I am doing this because:

- We are getting inconsistent behaviour from the Cookies banner with it reappearing some of the time. I've not been able to identify the source of the bug but the existing code is hard to follow and there are multiple points reading and writing from the cookie. Hopefully this rewrite will solve the problem but if it doesn't, it should make it _much_ simpler to debug and resolve.

### Follow on PRs
[ ] There will be a follow on PR to drop the existing cookies controller+helper code but I'm rolling it out in two deploys to avoid causing 404s for users - see #208 
[ ] The tables listing cookies do not show correctly on mobile at present. the is an existing issue - I'll add the new responsive functionality from the Rules of Origin PRs once those are merged - see #216

### CODE REVIEW

File: app/controllers/cookies/policies_controller.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+module Cookies
+  class PoliciesController < ApplicationController
+    before_action { @no_shared_search = true }
+
+    def show
+      @policy = cookies_policy
```
Comment (willfish) [inline app/controllers/cookies/policies_controller.rb]:
```suggestion
      @cookies_policy = cookies_policy
```

File: app/controllers/cookies/policies_controller.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    end
+
+    def create
+      @policy = cookies_policy
```
Comment (willfish) [inline app/controllers/cookies/policies_controller.rb]:
```suggestion
      @cookies_policy = cookies_policy
```

File: app/controllers/cookies/policies_controller.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      @policy.attributes = policy_params
```
Comment (willfish) [inline app/controllers/cookies/policies_controller.rb]:
```suggestion
      @cookies_policy.attributes = policy_params
```

File: app/controllers/cookies/policies_controller.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+      cookies[:cookies_policy] = {
+        value: @policy.to_cookie,
```
Comment (willfish) [inline app/controllers/cookies/policies_controller.rb]:
```suggestion
        value: @cookies_policy.to_cookie,
```

File: app/controllers/cookies/policies_controller.rb [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        expires: 1.year.from_now,
+      }
+      @policy.mark_persisted!
```
Comment (willfish) [inline app/controllers/cookies/policies_controller.rb]:
```suggestion
      @cookies_policy.mark_persisted!
```

File: app/controllers/cookies/policies_controller.rb (hunk 1 above)
Comment (jebw) [reply app/controllers/cookies/policies_controller.rb]: This will actually clash with the @cookies_policy set in `ApplicationController#cookies_policy` helper method - since this is duplicating that assignment anyway I've replaced references to `@policy` with usage of the helper_method `cookies_policy`
